assembl/views/api2/discussion.py

```diff
@@ -501,13 +505,74 @@ def get_time_series_analytics(request):
     return csv_response([r._asdict() for r in results], format, fieldnames)
 
 
+@view_config(context=InstanceContext, name="extract_csv_taxonomy",
+             ctx_instance_class=Discussion, request_method='GET',
+             permission=P_DISC_STATS)
+def extract_taxonomy_csv(request):
+    import assembl.models as m
+    discussion = request.context._instance
+    db = discussion.db
+    extracts = db.query(m.Extract).all()
```

Comment:
This looks like you get all extracts of the database, but what you want here is only extracts for this selected discussion (otherwise no need to have this route accessible from a discussion instance)

Reply:
you're absolutelu right, I did it and it takes much less time to download the file